lib/matplotlib/tri/triinterpolate.py

```diff
@@ -761,23 +757,20 @@ def get_function_derivatives(self, alpha, J, ecc, dofs):
             [       -z_sq,  2.*x*z-z_sq],
             [     x*z-y*z,      x*y-y*z]])
         # Puts back dV in first apex basis
-        dV = _prod_vectorized(dV, _extract_submatrices(
-            self.rotate_dV, subtri, block_size=2, axis=0))
-
-        prod = _prod_vectorized(self.M, dV)
-        prod += _scalar_vectorized(E[:, 0, 0],
-                                   _prod_vectorized(self.M0, dV))
-        prod += _scalar_vectorized(E[:, 1, 0],
-                                   _prod_vectorized(self.M1, dV))
-        prod += _scalar_vectorized(E[:, 2, 0],
-                                   _prod_vectorized(self.M2, dV))
+        dV = dV @ _extract_submatrices(
+            self.rotate_dV, subtri, block_size=2, axis=0)
+
+        prod = (self.M @ dV
```

Comment:
Do we pay a performance hit for temporaries here?

Reply:
Given that it‘s not in a tight loop, it can‘t be too bad. Anyway it‘s the same as before.

Reply:
actually I initially thought that the temporaries would be the same but that's not the case; even if I doubt this really matters performance-wise I restored the old form.